### ISSUE TITLE
`unicorn.util.smartHttp` Errors ignored

### ISSUE THREAD

**Issue by Commandcracker** [unicornpkg/libunicornpkg#33]
**Describe the bug**
Errors from `unicorn.util.smartHttp` are not visible to the user,
also it says the package was successfully installed all tho it fails to download a file.

**Screenshots**
![Screenshot_20221207_214729](https://user-images.githubusercontent.com/49335821/206293941-da09abbb-7863-4aa0-8209-5eef178a8233.png)

**Comment by tomodachi94** [comment]
Yeah, I've been meaning to completely redo `unicorn.util.smartHttp`. Might tackle this today.
